### PR TITLE
chore: support public stages being deprecated

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

https://github.com/discord/discord-api-docs/commit/8b2b1559c4b82f23421a5d74c2caf7005f9157f5

EDIT:
Restructured deprecation to go with the new documentation below:
https://github.com/discord/discord-api-docs/commit/47c47f80ca865f50024acf62ab62465a52065514
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Victorsitou** [comment]
Don't you think changing the documentation of `StageChannel.create_instance` and `StageInstance.edit` is also necessary? 

I was testing this, and it still can be used, so I guess a documentation note is more than enough (I was thinking about a deprecation warning)

**Comment by onerandomusername** [comment]
Hm. I can add a documentation note, but a deprecation warning is also a good idea.

**Comment by Victorsitou** [comment]
> Hm. I can add a documentation note, but a deprecation warning is also a good idea.

Both are useful tbh, you can add both, any opinions from the other collaborators?

**Comment by onerandomusername** [comment]
I'll add the docs for now and the deprecation warning will be added later.

File: disnake/channel.py [hunk 1]
```diff
@@ -1362,6 +1362,9 @@ async def create_instance(
             The stage instance's topic.
         privacy_level: :class:`StagePrivacyLevel`
             The stage instance's privacy level. Defaults to :attr:`StagePrivacyLevel.guild_only`.
+
+            .. deprecated:: 2.4
+
```
Comment (Victorsitou) [inline disnake/channel.py:1367]: I might be wrong, but when I said that changing the documentation is needed, I thought about a note mentioning that only `guild_only` is supported.

Setting it to public works too, so I guess it's not that important, a deprecation warning if someone tries to set it to public would be nice.

Comment (shiftinv) [reply disnake/channel.py:1367]: I agree, the property itself doesn't seem to be deprecated, only the `PUBLIC` value is

Comment (Victorsitou) [reply disnake/channel.py:1367]: A deprecation warning (with `utils.warn_deprecated`) when someone tries to use `StagePrivacyLevel.public` might be good too.

File: disnake/stage_instance.py [hunk 2]
```diff
@@ -96,7 +96,9 @@ def _update(self, data: StageInstancePayload):
         self.id: int = int(data["id"])
         self.channel_id: int = int(data["channel_id"])
         self.topic: str = data["topic"]
-        self.privacy_level: StagePrivacyLevel = try_enum(StagePrivacyLevel, data["privacy_level"])
+        self.privacy_level: StagePrivacyLevel = try_enum(
+            StagePrivacyLevel, data.get("privacy_level", 2)
+        )
```
Comment (EQUENOS) [inline disnake/stage_instance.py:101]: We should probably make this a property in order to show a deprecation warning

File: disnake/stage_instance.py [hunk 3]
```diff
@@ -152,6 +181,9 @@ async def edit(
         if privacy_level is not MISSING:
             if not isinstance(privacy_level, StagePrivacyLevel):
                 raise InvalidArgument("privacy_level field must be of type PrivacyLevel")
+            warn_deprecated(
```
Comment (shiftinv) [inline disnake/stage_instance.py:184]: This should only warn when trying to set `privacy_level` to `StagePrivacyLevel.public`; the field itself isn't deprecated, just that specific value is, so there shouldn't be any deprecation warnings when accessing `privacy_level` either.

File: disnake/stage_instance.py [hunk 4]
```diff
@@ -108,11 +106,22 @@ def channel(self) -> Optional[StageChannel]:
         # the returned channel will always be a StageChannel or None
         return self._state.get_channel(self.channel_id)  # type: ignore
 
-    def is_public(self) -> bool:
-        """Whether the stage instance is public.
+    @property
+    def discoverable_disabled(self) -> bool:
+        """discoverable_disabled: :class:`bool`
+        Whether discoverability for the stage instance is disabled."""
```
Comment (shiftinv) [inline disnake/stage_instance.py:112]:
```suggestion
        """:class:`bool`: Whether discoverability for the stage instance is disabled."""
```

**Review by shiftinv — APPROVED** [review]

**Pull request merged**